### DISCUSSION TITLE
initial window title doesn't get set on foot

### DISCUSSION THREAD

**Discussion by charbelnicolas** [jarun/nnn#1918] [Q&A]
As it is right now nnn won't show which directory it is on in the title, it is only when you move to another directory that it will show.

For example, I launch nnn through my window manager shortcut (Super key + F), it will open on the home folder but the title will be blank, only when I navigate to another folder it will show in the title.

This does not happen if you launch it directly from the terminal.

**Comment by charbelnicolas** [comment]
Here is a video to make it more clear:

[https://drive.google.com/file/d/1zXJrKmoXYLIUq25uWsSMNf2usMiNbedR/view?usp=sharing](https://drive.google.com/file/d/1zXJrKmoXYLIUq25uWsSMNf2usMiNbedR/view?usp=sharing)

**Comment by N-R-K** [comment]
It works fine for me using `xterm -e nnn -x`. Maybe `foot` is overriding the title? Try some other terminals (xterm if possible) and see if you can reproduce or not.

**Reply by charbelnicolas** [reply]
Yeah, something wrong is going on with foot, I'll try to report it, thank you

**Reply by charbelnicolas** [reply]
Hello @N-R-K, sorry to bother you, did you manage to figure out why it's popping the title on the foot terminal?

**Reply by N-R-K** [reply]
> did you manage to figure out why it's popping the title on the foot terminal?

I didn't have time for it, so no. And I see it as a minor issue so it's not a priority for me either.

**Reply by N-R-K** [reply]
> Hello @N-R-K, sorry to bother you, did you manage to figure out why it's popping the title on the foot terminal?

Looked at it again, but no, I still do not see where the pop is coming from. And I cannot reproduce it on my xterm either so it's almost impossible to debug. Does the following patch make any difference? (I don't think it will, but w/e).

```diff
                /* Save terminal window title */
-               printf("\033[22;0t");
+               printf("\033[22;2t");
```

**Reply by charbelnicolas** [reply]
> > Hello @N-R-K, sorry to bother you, did you manage to figure out why it's popping the title on the foot terminal?
> 
> Looked at it again, but no, I still do not see where the pop is coming from. And I cannot reproduce it on my xterm either so it's almost impossible to debug. Does the following patch make any difference? (I don't think it will, but w/e).
> 
> ```diff
>                 /* Save terminal window title */
> -               printf("\033[22;0t");
> +               printf("\033[22;2t");
> ```

I'm sorry, I don't use nnn anymore :\

**Reply by MatthiasKunnen** [reply]
~@N-R-K, I tested your patch, and unfortunately, it does not fix the issue as you expected.~
Disregard this, my nnn does not even update foot's title on directory change so I'm probably not even testing this issue properly.

I might get back to this later, time permitting.

**Reply by N-R-K** [reply]
> my nnn does not even update foot's title on directory change

You need to use the -x flag and have nnn compiled with "X11" support (should be enabled by default in the Makefile). E.g `xterm -e nnn -x` with xterm. (idk how foot expects initial program to run, for xterm it's `-e`).

**Reply by MatthiasKunnen** [reply]
It's `foot nnn -x`. I tested again with your patch and have to sadly report that it did not fix the issue. Only after a directory change is the title correct.

As a workaround, `foot true && nnn -x` does appear to work correctly both with and without the patch.

**Reply by charbelnicolas** [reply]
> It's `foot nnn -x`. I tested again with your patch and have to sadly report that it did not fix the issue. Only after a directory change is the title correct.
> 
> As a workaround, `foot true && nnn -x` does appear to work correctly both with and without the patch.

That doesn't work at all on my setup :)

**Comment by charbelnicolas** [comment]
Apparently according to foot's developer it's not a bug with foot

[https://codeberg.org/dnkl/foot/issues/1783](https://codeberg.org/dnkl/foot/issues/1783)

**Comment by izmyname** [comment]
I'm using this function in my zsh config to make foot update the window title. It works with nnn, as well.

```
# Called before prompt(?)
function precmd {
    # Set window title
    print -Pn "\e]0;%(1j,%j job%(2j|s|); ,)%~\e\\"
}

# Called when executing a command
function preexec {
    print -Pn "\e]0;${(q)1}\e\\"
}
```
